### ISSUE TITLE
Consumable system features

### ISSUE THREAD

**Issue by ledettwy** [NixOS/hydra#585]
Hydra clone of https://github.com/NixOS/nix/issues/2307, since Hydra has its own scheduler independent of nix.

I've already started working on this, but I'm not terribly well versed in C++, so definitely feel free to preempt me on this ^^

**Comment by edolstra** [comment]
Yeah, this is probably a lot easier to implement in Hydra than in Nix since the state is kept entirely in memory.
